.devcontainer/devcontainer.json

```diff
@@ -38,6 +47,50 @@
       "!include_dir_list scalar",
       "!include_dir_merge_list scalar",
       "!include_dir_merge_named scalar"
+    ],
+    // python common
+    "[python]": {
+      "editor.defaultFormatter": "ms-python.python",
+      "editor.codeActionsOnSave": {
+        "source.organizeImports": true
+      }
+    },
+    "python.linting.enabled": true,
+    "python.pythonPath": "/usr/local/bin/python",
+    // python linting
+    "python.linting.banditEnabled": true,
+    "python.linting.banditArgs": [
+      "--format=custom",
+      "--configfile=tests/bandit.yaml"
+    ],
+    "python.linting.flake8Enabled": true,
+    "python.linting.flake8Args": ["--config=./setup.cfg"],
+    "python.linting.mypyEnabled": true,
+    "python.linting.mypyArgs": [
+      "--follow-imports=silent",
+      "--ignore-missing-imports",
+      "--show-column-numbers",
+      "--no-pretty",
+      "--config-file=./mypy.ini"
+    ],
+    "python.linting.pylintEnabled": false,
+    "python.linting.pylamaEnabled": false,
+    "python.formatting.provider": "black",
+    "python.formatting.blackArgs": [
```

Comment:
So, this is for all of your comments you made about this since they are all the same,

_yes_ these are not technically needed. They are the default settings. But the way VS Code settings hierarchy works you should have them. Settings go Default -> User -> Devcontainer -> Workspace -> Folder. If user defines a user setting for Python, black, isort, etc. it will cause their devcontainer to use their user setting for it, not the default. Redefining them in the devcontainer override their user setting. So while redefining these setting seems unnecessary, it is basically just re-affirming the defaults for the devcontainer so the user settings are ignored.

Reply:
https://code.visualstudio.com/docs/getstarted/settings#_settings-precedence

Reply:
It took a bit to find the actual docs on it (I just knew the order because I use VS Code way too much 😂 ). But the full order would be 

1. Default
2. User
3. Remote (devcontainer/SSH)
4. Workspace (`.code-workspace` files)
5. Workspace folder settings (`.vscode/settings.json`)
6. Repeat for language specific blocks (`[python]` blocks)